pkg/forklift-api/webhooks/mutating-webhook/mutators/secret-mutator.go

```diff
@@ -38,14 +40,17 @@ func (mutator *OvirtCertMutator) Mutate(ar *admissionv1.AdmissionReview) *admiss
 	}
 
 	var insecure = false
-	if insecureSkipVerify, ok := secret.Data["insecureSkipVerify"]; ok {
-		insecure, err = strconv.ParseBool(string(insecureSkipVerify))
-		if err != nil {
-			log.Error(err, "Failed to parse insecure property from the secret")
-			return util.ToAdmissionResponseError(err)
+	if _, ok := secret.GetLabels()["createdForProviderType"]; ok {
+		if insecureSkipVerify, ok := secret.Data["insecureSkipVerify"]; ok {
+			insecure, err = strconv.ParseBool(string(insecureSkipVerify))
+			if err != nil {
+				log.Error(err, "Failed to parse insecure property from the secret")
+				return util.ToAdmissionResponseError(err)
+			}
+		} else {
+			secret.Data["insecureSkipVerify"] = []byte("false")
+			secretChanged = true
 		}
-	} else {
-		secret.Data["insecureSkipVerify"] = []byte("false")
 	}
 
 	if providerType, ok := secret.GetLabels()["createdForProviderType"]; ok && providerType == "ovirt" && !insecure {
```

Comment:
unrelated to this PR but here we can also assume the `createdForProviderType` label exists

Reply:
right, the label still need to be fetch for the ovirt provider check, but the OK check is redundant.